From a8431d1f972333348493e2efcf9f0da44fdb5add Mon Sep 17 00:00:00 2001
From: Trevor Tao <trevor.tao@arm.com>
Date: Fri, 12 Nov 2021 23:37:03 +0800
Subject: Rebuilding docker-builder if code changed (#36042)

Rebuilding docker-builder if code changed, especially for
development phase.

Signed-off-by: trevor.tao <trevor.tao@arm.com>
---
 tools/docker | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/tools/docker b/tools/docker
index 33ec08adfc..0152231f30 100755
--- a/tools/docker
+++ b/tools/docker
@@ -32,7 +32,8 @@ export REPO_ROOT="${ROOT}"
 if [[ -f out/${TARGET_OS}_${TARGET_ARCH}/docker-builder ]]; then
   CURRENT_BUILD="$(./common/scripts/report_build_info.sh | grep buildGitRevision | cut -d= -f2)"
   PREBUILT="$(out/${TARGET_OS}_${TARGET_ARCH}/docker-builder --version)"
-  if [[ "${CURRENT_BUILD}" != "${PREBUILT}" ]]; then
+  CODE_CHG="$(git diff ./tools/docker-builder)"
+  if [[ "${CURRENT_BUILD}" != "${PREBUILT}" ]] || [[ -n "${CODE_CHG}" ]]; then
     GOOS=${GOOS_LOCAL} GOARCH=${GOARCH_LOCAL} ./common/scripts/gobuild.sh out/${TARGET_OS}_${TARGET_ARCH}/docker-builder ./tools/docker-builder
   fi
 else
-- 
2.35.3

